### PR TITLE
OSDOCS-1688 - Adding assembly on validating an installation

### PR DESCRIPTION
**Pull request by pneedle-rh**
This applies to master, branch/enterprise-4.6 and branch-[enterprise-4](https://issues.redhat.com/browse/enterprise-4).7.

The focus of this PR is to help enhance the end-to-end reader journey through the docs when installing OCP on AWS by using user-provisioned infrastructure.

This relates to [OSDOCS-1688 - [Docs] Create a 'Validating an installed platform' assembly](https://issues.redhat.com/browse/OSDOCS-1688).

The preview is here: https://osdocs-1688-validating-an-installation--ocpdocs.netlify.app/openshift-enterprise/latest/installing/validating-an-installation.html

### CODE REVIEW

**Comment by pneedle-rh** [comment]
@openshift/team-documentation for peer review please.

File: installing/validating-an-installation.adoc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+You can check the status of an {product-title} cluster after an installation by following the procedures in this document.
+
+//Reviewing the installation log
+include::modules/reviewing-the-installation-log.adoc[leveloffset=+1]
+
+//Getting cluster version and update details
+include::modules/getting-cluster-version-and-update-details.adoc[leveloffset=+1]
+
+.Additional resources
```
Comment (jboxman) [inline installing/validating-an-installation.adoc:16]: For assemblies, I believe we're doing `==` headers along with an `[id=""]` instead of `.`.

Comment (jboxman) [reply installing/validating-an-installation.adoc:16]: But this is a cheat because of the lack of `xref` support? Ignore the previous comment, then.

File: installing/validating-an-installation.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+* See xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[Understanding the monitoring stack] for more information about the {product-title} monitoring stack.
+
+//Listing alerts that are firing
+include::modules/listing-alerts-that-are-firing.adoc[leveloffset=+1]
+
+.Additional resources
+
+* See xref:../monitoring/managing-alerts.adoc#managing-alerts[Managing alerts] for further details about alerting in {product-title}.
+
+== Next steps
```
Comment (jboxman) [inline installing/validating-an-installation.adoc:57]: Include an `[id=""]` here.

File: installing/validating-an-installation.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+* xref:../support/troubleshooting/troubleshooting-installations.adoc#troubleshooting-installations[Troubleshooting installations]
```
Comment (jboxman) [inline installing/validating-an-installation.adoc:59]: Is this actually a required next step?

Or maybe preface with _Optional:_

File: modules/checking-cluster-resource-availability-and-utilisation.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="checking-cluster-resource-availability-and-utilisation_{context}"]
+= Checking cluster resource availability and utilisation
```
Comment (jboxman) [inline modules/checking-cluster-resource-availability-and-utilisation.adoc:6]: s/utilisation/utilization/

File: modules/checking-cluster-resource-availability-and-utilisation.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+* etcd
+
+* Kubernetes compute resources
+
+* Kubernetes network resources
+
+* Prometheus
+
+* USE method dashboards relating to cluster and node performance
```
Comment (jboxman) [inline modules/checking-cluster-resource-availability-and-utilisation.adoc:20]: What is a _USE_ method?

File: modules/listing-alerts-that-are-firing.adoc [hunk 6]
```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="listing-alerts-that-are-firing_{context}"]
+= Listing alerts that are firing
+
+Alerts provide notifications that a set of circumstances are apparent within an {product-title} cluster. You can review the alerts that are firing in your cluster by using the Alerting UI in the {product-title} web console.
```
Comment (jboxman) [inline modules/listing-alerts-that-are-firing.adoc:8]: Or,

> Alerts provide notifications about exceptional events within an {product-title} cluster.

File: modules/listing-alerts-that-are-firing.adoc [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. In the *Administrator* perspective, navigate to the *Monitoring* -> *Alerting* -> *Alerts* page.
+
+. Review the alerts that are firing, including their *Severity*, *State* and *Source*.
```
Comment (jboxman) [inline modules/listing-alerts-that-are-firing.adoc:18]: > *State*, and

We serial comma.

File: modules/logging-in-by-using-the-web-console.adoc [hunk 8]
```diff
@@ -0,0 +1,47 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="logging-in-by-using-the-web-console_{context}"]
+= Logging in to the cluster by using the web console
+
+You can log into your cluster as `kubeadmin` by using the {product-title} web console after the Console Operator becomes available.
```
Comment (jboxman) [inline modules/logging-in-by-using-the-web-console.adoc:8]: What's `kubeadmin`?

> You can log into your cluster as the `kubeadmin` user...

File: modules/logging-in-by-using-the-web-console.adoc [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+.Prerequisites
+
+* You have access to the installation host.
+* The Console Operator is available on your cluster.
```
Comment (jboxman) [inline modules/logging-in-by-using-the-web-console.adoc:13]: Would the Console Operator ever not be installed?

File: modules/logging-in-by-using-the-web-console.adoc [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+.Procedure
+
+. Obtain the `kubeadmin` password from the `kubeadmin-password` file on the installation host:
```
Comment (jboxman) [inline modules/logging-in-by-using-the-web-console.adoc:17]: `kubeadmin` user?

File: modules/logging-in-by-using-the-web-console.adoc [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+Alternatively, you can obtain the {product-title} route from the `<install_dir>/.openshift_install.log` log file on the installation host.
+====
++
+.Example output
+[source,terminal]
+----
+console     console-openshift-console.apps.<cluster_name>.<base_domain>            console     https   reencrypt/Redirect   None
+----
+
+. Navigate to the route detailed in the output of the preceding command in a web browser and log in as `kubeadmin`.
```
Comment (jboxman) [inline modules/logging-in-by-using-the-web-console.adoc:47]: > the `kubeadmin` user.

File: modules/reviewing-the-installation-log.adoc [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
++
+Cluster credentials are included at the end of the log if the installation is successful, as outlined in the following example:
++
+[source,terminal]
+----
+...
+time="2020-12-03T09:50:47Z" level=info msg="Install complete!"
+time="2020-12-03T09:50:47Z" level=info msg="To access the cluster as the system:admin user when using 'oc', run 'export KUBECONFIG=<install_dir>/auth/kubeconfig'"
+time="2020-12-03T09:50:47Z" level=info msg="Access the OpenShift web-console here: https://console-openshift-console.apps.<cluster_name>.<base_domain>"
+time="2020-12-03T09:50:47Z" level=info msg="Login to the console with user: \"kubeadmin\", and password: \"<password>\""
```
Comment (jboxman) [inline modules/reviewing-the-installation-log.adoc:33]: I don't know if saying anything about `<password>` is needed here or not. I assume it's replaced with a random password, generated by OCP?

**Review by jboxman — CHANGES_REQUESTED** [review]
I've left a few comments for your review. Thanks!

File: installing/validating-an-installation.adoc (hunk 1 above)
Comment (pneedle-rh) [reply installing/validating-an-installation.adoc:16]: Yes this is an 'Additional resources' section specific to the module that precedes this line.

File: installing/validating-an-installation.adoc (hunk 2 above)
Comment (pneedle-rh) [reply installing/validating-an-installation.adoc:57]: Thank you for highlighting this. Resolved.

File: installing/validating-an-installation.adoc (hunk 3 above)
Comment (pneedle-rh) [reply installing/validating-an-installation.adoc:59]: That is a good point. I have subsequently updated that line as follows:

* See xref:../support/troubleshooting/troubleshooting-installations.adoc#troubleshooting-installations[Troubleshooting installations] if you experience issues when installing your cluster.

File: modules/checking-cluster-resource-availability-and-utilisation.adoc (hunk 4 above)
Comment (pneedle-rh) [reply modules/checking-cluster-resource-availability-and-utilisation.adoc:6]: Thank you. I have now resolved this in the title and in the module name.

File: modules/checking-cluster-resource-availability-and-utilisation.adoc (hunk 5 above)
Comment (pneedle-rh) [reply modules/checking-cluster-resource-availability-and-utilisation.adoc:20]: It is a method used when approaching performance issues. It stands for 'utilization, saturation and errors'. My understanding is that the cluster and node performance dashboards provided statistics relating to each of those areas. For the purposes of this bulleted list, I think that a brief summary is suffice. I have subsequently changed that bullet point to the following:

* Dashboards relating to cluster and node performance

File: modules/listing-alerts-that-are-firing.adoc (hunk 6 above)
Comment (pneedle-rh) [reply modules/listing-alerts-that-are-firing.adoc:8]: My view is that the events might not be exceptional. Some alerts are informational. Alerts fire when explicit conditions defined in alerting rules are true for a defined time period. I have updated this sentence to the following:

"Alerts provide notifications that a set of defined conditions are true in an {product-title} cluster."

Comment (pneedle-rh) [reply modules/listing-alerts-that-are-firing.adoc:8]: There is also an 'Additional resources' link following this module to the [Managing alerts](https://docs.openshift.com/container-platform/4.6/monitoring/managing-alerts.html) section so that readers can learn more about alerting.

File: modules/listing-alerts-that-are-firing.adoc (hunk 7 above)
Comment (pneedle-rh) [reply modules/listing-alerts-that-are-firing.adoc:18]: Good catch! Resolved.

File: modules/logging-in-by-using-the-web-console.adoc (hunk 8 above)
Comment (pneedle-rh) [reply modules/logging-in-by-using-the-web-console.adoc:8]: Thank you for highlighting this. I have subsequently updated the introduction to this module as follows:

"The `kubeadmin` user exists by default after an {product-title} installation. You can log into your cluster as the `kubeadmin` user by using the {product-title} web console after the Console Operator becomes available."

File: modules/logging-in-by-using-the-web-console.adoc (hunk 9 above)
Comment (pneedle-rh) [reply modules/logging-in-by-using-the-web-console.adoc:13]: Yes. After a UPI installation there might be a requirement to approve pending certificate signing requests. Until those are approved, the Console Operator might be unavailable. If that is the case, then the console will not be available. I experienced this in my testing.

Thank you for highlighting this, I have made need for the prerequisite clearer by adding the following note after that line:

[NOTE]
====
You might need to approve some certificate signing requests (CSRs) after a user-provisioned infrastructure installation. The Console Operator might be unavailable until some CSRs are approved. You cannot access the {produtct-title} web console unless the Console Operator is available.
====

File: modules/logging-in-by-using-the-web-console.adoc (hunk 10 above)
Comment (pneedle-rh) [reply modules/logging-in-by-using-the-web-console.adoc:17]: Thank you for highlighting this. I have subsequently changed this line to:

. Obtain the password for the `kubeadmin` user from the `kubeadmin-password` file on the installation host:

File: modules/logging-in-by-using-the-web-console.adoc (hunk 11 above)
Comment (pneedle-rh) [reply modules/logging-in-by-using-the-web-console.adoc:47]: Resolved.

File: modules/reviewing-the-installation-log.adoc (hunk 12 above)
Comment (pneedle-rh) [reply modules/reviewing-the-installation-log.adoc:33]: Thank you for highlighting this. I have subsequently updated the example output to include an example password. I have also replaced `<cluster_name>.<base_domain>` with `mycluster.example.com`. I noticed that `example.com` is used as an example base domain elsewhere in the installation book.

File: modules/getting-cluster-version-and-update-details.adoc [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+[source,terminal]
+----
+$ oc describe clusterversion
+----
+
+. List the current update channel:
++
+[source,terminal]
+----
+$ oc get clusterversion -o json|jq ".items[0].spec"
```
[the remark this answers is not in the document]
Comment (pneedle-rh) [reply modules/getting-cluster-version-and-update-details.adoc]: Thank you for highlighting this. I have replaced the command and output accordingly.

**Comment by pneedle-rh** [comment]
@staebler can you please let me know who I should request SME review from, for this PR? Thanks!

**Comment by pneedle-rh** [comment]
@jianlinliu can you please let me know who I should request QE review from, for this PR? Thanks!

File: modules/logging-in-by-using-the-web-console.adoc (hunk 9 above)
Comment (kalexand-rh) [reply modules/logging-in-by-using-the-web-console.adoc:13]: I'd probably make the prereq something like, "You have completed cluster installation, and all cluster operators are available." I agree that adding the note with more context would be confusing for the IPI installations, but that a little more info might be useful for UPI. (And that if we need to add more info about the CSRs and core Operator availability, that the UPI installation modules/assemblies themselves are the better place for it.)

Comment (pneedle-rh) [reply modules/logging-in-by-using-the-web-console.adoc]: That's good advice, thank you!

I have a task in my proposal document to add a note in the 'Approving the CSRs for your machines' section in the UPI installation procedure, which will meet this requirement.

I have now updated the intro to this module to remove the reference to the console operator specifically. I have also changed the prerequisite accordingly.

**Review by jboxman — APPROVED** [review]

**Comment by jianlinliu** [comment]
LGTM.

File: modules/getting-cluster-version-and-update-details.adoc [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
++
+[source,terminal]
+----
+$ oc get clusterversion
+----
++
+.Example output
+[source,terminal]
+----
+Cluster version is 4.6.4
```
Comment (staebler) [inline modules/getting-cluster-version-and-update-details.adoc:27]: Is this intentionally a stripped down version of the full output?

Example,
```
$ oc get clusterversion
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.6.4     True        False         3m26s   Cluster version is 4.6.4
```

File: modules/querying-the-status-of-cluster-components-using-the-cli.adoc [hunk 15]
```diff
@@ -0,0 +1,131 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="querying-the-status-of-cluster-components-using-the-cli_{context}"]
+= Querying the status of cluster components by using the CLI
+
+You can verify the status of certificate signing requests (CSRs), cluster Operators, nodes, and the pods running in core {product-title} projects.
```
Comment (staebler) [inline modules/querying-the-status-of-cluster-components-using-the-cli.adoc:8]: I am surprised that we are directing users to do these things as part of validating an installation.
1) I would not expect the user to have to look at CSRs at all unless the user is trying to diagnose why the cluster does not have the nodes that the user expects.
2) I would not expect the user to look at ClusterOperator resources directly. I would expect the user to look at the aggregated status in the ClusterVersion resource to ascertain whether there are cluster operators that are not available.
3) I would not expect the user to look at pods unless the user is trying to diagnose why a specific ClusterOperator is not available.

In other words, as part of validating an installation, I would only expect the user to look at ClusterVersion and Nodes. If either of those are showing problems, then looking at things like CSRs, ClusterOperators, and Pods may be something that they would then do, but that would be part of troubleshooting rather than validating.

File: modules/getting-cluster-version-and-update-details.adoc (hunk 14 above)
Comment (pneedle-rh) [reply modules/getting-cluster-version-and-update-details.adoc]: @staebler thank you for your review.

Thanks for highlighting this. The intention was to include the full output. I have now resolved this.

File: modules/querying-the-status-of-cluster-components-using-the-cli.adoc (hunk 15 above)
Comment (pneedle-rh) [reply modules/querying-the-status-of-cluster-components-using-the-cli.adoc]: @staebler thank you for your guidance. I have subsequently made the following changes to this PR:

- I have updated the title for this section and the content so that it focuses on the nodes.

- I have updated the [Getting cluster version and update details](https://osdocs-1688-validating-an-installation--ocpdocs.netlify.app/openshift-enterprise/latest/installing/validating-an-installation.html#getting-cluster-version-and-update-details_validating-an-installation) section of this document to include a step stating that if the cluster status indicates that the installation is still progressing, you can obtain more detailed progress information by checking the status of the Operators.

- I have included a new additional resources link after the [Getting cluster version and update details](https://osdocs-1688-validating-an-installation--ocpdocs.netlify.app/openshift-enterprise/latest/installing/validating-an-installation.html#getting-cluster-version-and-update-details_validating-an-installation) section, linking to the [Querying Operator status after installation](https://osdocs-1688-validating-an-installation--ocpdocs.netlify.app/openshift-enterprise/latest/support/troubleshooting/troubleshooting-installations.html#querying-operator-status-after-installation_troubleshooting-installations) section of the Troubleshooting Guide.

- I have updated the [Querying Operator status after installation](https://osdocs-1688-validating-an-installation--ocpdocs.netlify.app/openshift-enterprise/latest/support/troubleshooting/troubleshooting-installations.html#querying-operator-status-after-installation_troubleshooting-installations) section of the Troubleshooting Guide to include some details about checking that CSRs are approved.

File: modules/querying-the-status-of-cluster-nodes-using-the-cli.adoc [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+[source,terminal]
+----
+$ oc get nodes
+----
++
+.Example output
+[source,terminal]
+----
+NAME                        STATUS   ROLES    AGE   VERSION
+ip-<ip_address>.internal    Ready    worker   33m   v1.19.0+9f84db3
```
Comment (staebler) [inline modules/querying-the-status-of-cluster-nodes-using-the-cli.adoc:28]: I don't know that the node name typically has this format. I usually see the node name as having the hostname of the machine.

**Review by staebler — COMMENTED** [review]
This looks good to me, save for a few minor things.

File: modules/querying-the-status-of-cluster-nodes-using-the-cli.adoc (hunk 16 above)
Comment (pneedle-rh) [reply modules/querying-the-status-of-cluster-nodes-using-the-cli.adoc]: @staebler thank you for highlighting this. Some provider-formatted host names include IPv4 addresses. Since the host names in this assembly do not need to be provider-specific, I have updated the examples with more generic names that do not include IP addresses.

File: modules/querying-operator-status-after-installation.adoc [hunk 17]
```diff
@@ -21,7 +21,52 @@ You can check Operator status at the end of an installation. Retrieve diagnostic
...
+<1> A client request CSR.
+<2> A server request CSR.
++
+In this example, two machines are joining the cluster. You might see more approved CSRs in the list.
+
+.. If the CSRs were not approved, after all of the pending CSRs for the machines you added are in `Pending` status, approve the CSRs for your cluster machines:
++
+[NOTE]
+====
+Because the CSRs rotate automatically, approve your CSRs within an hour of adding the machines to the cluster. If you do not approve them within an hour, the certificates will rotate, and more than two certificates will be present for each node. You must approve all of these certificates. After you approve the initial CSRs, the subsequent node client CSRs are automatically approved by the cluster `kube-controller-manager`. You must implement a method of automatically approving the kubelet serving certificate requests.
```
[the remark this answers is not in the document]
Comment (pneedle-rh) [reply modules/querying-operator-status-after-installation.adoc]: @staebler following detailed conversations with @sjenning and @kalexand-rh, I have created these two Bugzillas in relation to this comment:

* [Docs BZ 1914769](https://bugzilla.redhat.com/show_bug.cgi?id=1914769)
* [Engineering BZ 1914773](https://bugzilla.redhat.com/show_bug.cgi?id=1914773)

For the short term I have created [a new note in this PR](https://github.com/openshift/openshift-docs/pull/27823/files#diff-f0ffa2b36d6aeb62b9f56aef924dad3354ad2240bbb81066d3d7554a83299932R54-R57) so that the requirement and impact are clearer in the doc, at a high level. [The note in the original 'Approving the certificate signing requests for your machines' section](https://github.com/openshift/openshift-docs/pull/27823/files#diff-b87e23ae97d2c1cbe9acaeea182675205279379acf45a75caf2cc94ba04f0a34R99-R102) has also been updated with the same change in this PR.

For the longer term [BZ 1914773](https://bugzilla.redhat.com/show_bug.cgi?id=1914773) is a requests for our Engineering team to create and maintain an example script for OCP 4 that satisfies this requirement. We can then embed that script into a new procedure in the documentation that describes how to set this up in detail.

Please let me know if the note that I have included in the PR is sufficient as a short term solution. Thanks!

@jianlinliu can you please review this new note from a QE perspective? Thanks!

@kalexand-rh can you please peer review this new note from a docs perspective? Thanks!

File: modules/querying-operator-status-after-installation.adoc [hunk 18]
```diff
@@ -21,7 +21,57 @@ You can check Operator status at the end of an installation. Retrieve diagnostic
 $ oc get clusteroperators
 ----
 
-. If any Operators fail to become available, view Operator events:
+. Verify that all of the required certificate signing requests (CSRs) are approved. Some cluster Operators might not become available if there are pending CSRs.
```
Comment (jianlinliu) [inline modules/querying-operator-status-after-installation.adoc:24]: csr issues will lead to some unready nodes, or not found some expected nodes, I do not think they should be linked clusteroperators troubleshooting.

File: installing/validating-an-installation.adoc [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+include::modules/reviewing-the-installation-log.adoc[leveloffset=+1]
+
+//Getting cluster version, status, and update details
+include::modules/getting-cluster-version-status-and-update-details.adoc[leveloffset=+1]
+
+.Additional resources
+
+* See xref:../support/troubleshooting/troubleshooting-installations.adoc#querying-operator-status-after-installation_troubleshooting-installations[Querying Operator status after installation] for more information about querying Operator status if your installation is still progressing.
+
+* See xref:../support/troubleshooting/troubleshooting-operator-issues.adoc#troubleshooting-operator-issues[Troubleshooting Operator issues] for information about investigating issues with cluster Operators.
```
Comment (jianlinliu) [inline installing/validating-an-installation.adoc:20]: "cluster Operators"? Form the link, it is talking about application operators. And from its naming, this make me feel this item duplicated with the last one, the last link (Querying Operator status after installation) sounds more reasonable for cluster operator trouble shooting.

Comment (pneedle-rh) [reply installing/validating-an-installation.adoc:20]: @jianlinliu thank you for your further review.

I have updated this line to state "...investigating issues with cluster Operators". I think it is important to keep this additional reference in place, because it points to a useful assembly on how to address issues with operators. If the reader finds operator problems when validating an installation, it is important that this relevant troubleshooting assembly is found.

I don't think it duplicates the previous additional resource, since that one is a link to a different assembly in the troubleshooting guide, that explicitly talks about querying operator status after installation. My view is that they compliment each other are are both relevant.

"Operator" is always capitalised in our documentation.

File: modules/querying-operator-status-after-installation.adoc (hunk 18 above)
Comment (pneedle-rh) [reply modules/querying-operator-status-after-installation.adoc:24]: @jianlinliu in my test UPI installation for OCP on AWS, the Console Operator and others did not become active until I approved pending CSRs. This prevented me from being able to see the OCP web console. I have included this here because it would be useful for the reader to check that CSRs are approved in this type of scenario.

Please let me know if this makes sense.

Comment (jianlinliu) [reply modules/querying-operator-status-after-installation.adoc:24]: From my understanding, not approved csr -> node not ready -> some cluster operator does not get ready (not only console operator). So before check cluster operators, firstly need ensure all nodes get ready.

Comment (pneedle-rh) [reply modules/querying-operator-status-after-installation.adoc]: Thank you for clarifying that. Since this module is specifically about troubleshooting operator issues after an installation, the CSR approval steps are still relevant here. From your comment my understanding is still that some cluster operators do not become ready if some CSRs are not approved. I have updated the step to mention that nodes might not be ready also, for context:

". Verify that all of the required certificate signing requests (CSRs) are approved. Some nodes might not move to a `Ready` status and some cluster Operators might not become available if there are pending CSRs."

Does this meet your requirement?

File: modules/installation-approve-csrs.adoc [hunk 20]
```diff
@@ -93,7 +93,12 @@ endif::ibm-z[]
...
+Because the CSRs rotate automatically, approve your CSRs within an hour of adding the machines to the cluster. If you do not approve them within an hour, the certificates will rotate, and more than two certificates will be present for each node. You must approve all of these certificates. After you approve the initial CSRs, the subsequent node client CSRs are automatically approved by the cluster `kube-controller-manager`.
+====
++
+[NOTE]
+====
+For clusters running on platforms that are not machine API enabled, you must implement a method of automatically approving the kubelet serving certificate requests (CSRs). If a request is not approved, then `oc exec`, `oc rsh`, and `oc logs` commands will not succeed, because the serving certificate is required when the API server connects to the kubelet. Any operation that contacts the Kubelet endpoint requires this certificate approval to be in place. The method must watch for new CSRs, confirm that the CSR was submitted by the `node-bootstrapper` service account in the `system:node` or `system:admin` groups, and confirm the identity of the node.
```
Comment (kalexand-rh) [inline modules/installation-approve-csrs.adoc:101]:
```suggestion
For clusters running on platforms that are not machine API enabled, such as bare metal, you must implement a method of automatically approving the kubelet serving certificate requests (CSRs). If a request is not approved, then the `oc exec`, `oc rsh`, and `oc logs` commands cannot succeed because the serving certificate is required when the API server connects to the kubelet. Any operation that contacts the Kubelet endpoint requires this certificate approval to be in place. The method must watch for new CSRs, confirm that the CSR was submitted by the `node-bootstrapper` service account in the `system:node` or `system:admin` groups, and confirm the identity of the node.
```

Comment (pneedle-rh) [reply modules/installation-approve-csrs.adoc:101]: Thank you. I have subsequently updated this note and the one in `modules/querying-operator-status-after-installation.adoc`.

**Comment by jianlinliu** [comment]
lgtm.

**Comment by pneedle-rh** [comment]
@kalexand-rh can you please let me know if this PR now looks good from a peer review perspective? Thanks!

**Comment by staebler** [comment]
lgtm

**Comment by pneedle-rh** [comment]
@openshift/team-documentation for peer review please. This PR has been peer reviewed previously, but a number of changes have been applied since then and so another pass might be useful. Thanks!

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc [hunk 21]
```diff
@@ -0,0 +1,55 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="reviewing-cluster-status-from-the-openshift-cluster-manager_{context}"]
+= Reviewing the cluster status from the {cloud-redhat-com}
+
+You can review detailed information about the status of your cluster in the {cloud-redhat-com}.
```
Comment (bobfuru) [inline modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:8]: Should we drop the second "the" in this sentence. For example, we don't say "the OpenShift Container Platform", but not sure if the same should apply for Cluster Manager.

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
++
+* The cluster ID, status, type, location, and the provider name
++
+* Node counts by node type
++
+* Cluster version details, the creation date of the cluster, and the name of the cluster owner
++
+* The life cycle support status of the cluster
++
+* Subscription information including the service level agreement (SLA) status, the subscription unit type, the production status of the cluster, the subscription obligation and the service level
```
Comment (bobfuru) [inline modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:35]: s/Subscription information including the service level agreement (SLA) status, the subscription unit type, the production status of the cluster, the subscription obligation and the service level/Subscription information, including the service level agreement (SLA) status, the subscription unit type, the production status of the cluster, the subscription obligation, and the service level

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
++
+* A cluster history
+
+. Navigate to the *Monitoring* page to review the following information:
+* A list of any issues that have been detected
++
+* A list of alerts that are firing
++
+* The Cluster Operator status and version
```
Comment (bobfuru) [inline modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:44]: "The cluster Operator" ?

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc [hunk 24]
```diff
@@ -0,0 +1,55 @@
...
++
+* Cluster resource usage
+
+. Navigate to the *Insights* page to review the following information that is provided by Red Hat Insights:
```
Comment (bobfuru) [inline modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:48]: I'd drop "that is" here. I know we often add "that" for clarification, but here it seems clunky to me.

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc [hunk 25]
```diff
@@ -0,0 +1,55 @@
...
+* Potential issues that your cluster might be exposed to, categorized by risk level
++
+* Health check status by category
```
Comment (bobfuru) [inline modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:51]: s/Health check status/Health-check status" to reduce ambiguity here

File: modules/checking-cluster-resource-availability-and-utilization.adoc [hunk 26]
```diff
@@ -0,0 +1,41 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="checking-cluster-resource-availability-and-utilization_{context}"]
+= Checking cluster resource availability and utilization
+
+{product-title} provides a comprehensive set of monitoring dashboards that help you understand the state of cluster components.
+
+In the *Administrator* perspective you can access dashboards for core {product-title} components, including:
```
Comment (bobfuru) [inline modules/checking-cluster-resource-availability-and-utilization.adoc:10]: s/perspective you can access/perspective, you can access/

File: modules/listing-alerts-that-are-firing.adoc [hunk 27]
```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="listing-alerts-that-are-firing_{context}"]
+= Listing alerts that are firing
+
+Alerts provide notifications that a set of defined conditions are true in an {product-title} cluster. You can review the alerts that are firing in your cluster by using the Alerting UI in the {product-title} web console.
```
Comment (bobfuru) [inline modules/listing-alerts-that-are-firing.adoc:8]: "Alerts provide notifications _that_ a set" --> "Alerts provide notifications _when_ a set" ?

Comment (bobfuru) [reply modules/listing-alerts-that-are-firing.adoc:8]: Also, should "Alerting UI" be in bold? Is it a menu option, tab label, or similar?

File: installing/validating-an-installation.adoc [hunk 28]
```diff
@@ -0,0 +1,62 @@
...
+.Additional resources
+
+* See xref:../monitoring/managing-alerts.adoc#managing-alerts[Managing alerts] for further details about alerting in {product-title}.
+
+[id="validating-an-installation-next-steps"]
+== Next steps
+
+* See xref:../support/troubleshooting/troubleshooting-installations.adoc#troubleshooting-installations[Troubleshooting installations] if you experience issues when installing your cluster.
+
+* xref:../post_installation_configuration/machine-configuration-tasks.adoc#machine-configuration-tasks[Post-installation machine configuration tasks]
```
Comment (bobfuru) [inline installing/validating-an-installation.adoc:62]: I think this should be written as a step, something like:
"To configure operating system features, see _Post-installation machine configuration tasks_."

**Review by bobfuru — COMMENTED** [review]
A few very minor nits but overall, this LGTM. Nicely done!!

File: installing/validating-an-installation.adoc (hunk 28 above)
Comment (pneedle-rh) [reply installing/validating-an-installation.adoc:62]: Thank you Bob for your review and for highlighting this. On reflection I have updated this next step to direct the reader to the 'Post-installation configuration -> Cluster tasks' assembly as follows:

* After installing {product-title}, you can further xref:../post_installation_configuration/cluster-tasks.adoc#post-install-cluster-tasks[customize your cluster] to your requirements.

File: modules/checking-cluster-resource-availability-and-utilization.adoc (hunk 26 above)
Comment (pneedle-rh) [reply modules/checking-cluster-resource-availability-and-utilization.adoc:10]: Resolved.

File: modules/listing-alerts-that-are-firing.adoc (hunk 27 above)
Comment (pneedle-rh) [reply modules/listing-alerts-that-are-firing.adoc:8]: I have resolved the first suggestion.

Regarding the second suggestion, the Alerting UI is a reference to *Monitoring* -> *Alerting* -> *Alerts*, but it is not a menu option, tab label or suchlike itself, as far as I know. The procedure highlights how to navigate to *Monitoring* -> *Alerting* -> *Alerts* and states those menu items in bold. As such, I will leave "Alerting UI" as it is.

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc (hunk 21 above)
Comment (pneedle-rh) [reply modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:8]: Looking through the OpenShift docs library, most references are to "the Red Hat OpenShift Cluster Manager". Given this, I will keep this line as it is for now. If we decide that the "the" should be dropped, we should do so throughout the library, in my view.

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc (hunk 22 above)
Comment (pneedle-rh) [reply modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:35]: Thank you for highlighting this. Resolved.

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc (hunk 23 above)
Comment (pneedle-rh) [reply modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:44]: Good catch! Resolved.

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc (hunk 24 above)
Comment (pneedle-rh) [reply modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:48]: Thank you for highlighting this. Resolved.

File: modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc (hunk 25 above)
Comment (pneedle-rh) [reply modules/reviewing-cluster-status-from-the-openshift-cluster-manager.adoc:51]: Resolved.
